### DISCUSSION TITLE
Signal Desktop

### DISCUSSION THREAD

**Discussion by CalfMoon** [catppuccin/catppuccin#2382] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the suggested port be themed?

I'm aware of the fact that #1703's idea was rejected stating that upstream is not friendly yet. Although upstream hasn't become more friendly, I've been working on theming signal on my own and [this](https://github.com/CalfMoon/signal-desktop) is my work so far. It looks like this:
![image](https://github.com/catppuccin/catppuccin/assets/111578326/5a7e6a34-a5fc-4e24-82d3-fa7f070f8150)
Yes, it is still not very easy to theme, but I've been working on another personal project that tries to make this process easier. Till then the manual process is the only way but for those who like to keep their desktop consistent at any cost this can be helpful. 
Also only mocha is being developed, but once its complete it will be very easy to copy the template.

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by unseen-ninja** [comment]
As stated in #1703 we're happy to reconsider this.. but since the conditions haven't changed I'm torn a little. While I think this is a nice port the application of it is really wonky. Let's see where things stand once the other flavours are also in place. We can always hope for Signal to make theming a thing, right … ? ![copium](https://cdn.discordapp.com/emojis/898969520989949974.webp?size=16)

**Reply by CalfMoon** [reply]
We can *hope* but I don't think it would be best to only rely on that. We don't know how far in time it will be implemented,  or even if it will be implemented.
By "application of it is really wonky" you mean the way we insert the theme, yes, it is wonky but like I said, I've been writing a theme-injector for signal. The script itself is mostly complete but I'm trying to package it properly right now.
I will update this discussion once other themes and my project are ready. Hopefully the discussion won't be closed by then.

**Reply by CalfMoon** [reply]
Sorry for being this late, I didn't work on this for this long but I got distracted with other things (games). Its still not perfect but i think its better than before. The weirdest one is latte as the color of text is still back in sent text as signal thinks the sent message has accent color. It is not a difficult fix but i realized that at the last moment when i was taking screenshot.
I  completed [another project](https://github.com/CalfMoon/signal-themer) that makes it easier to inject the theme.
![image](https://github.com/catppuccin/catppuccin/assets/111578326/620b99b6-85a2-4121-893e-7415e94a3231)
![image](https://github.com/catppuccin/catppuccin/assets/111578326/541057dc-c4b8-44f6-97d5-083da6d1dabd)
![image](https://github.com/catppuccin/catppuccin/assets/111578326/1b2f1192-cac0-4f39-b4e6-b5d5f0b8d164)
![image](https://github.com/catppuccin/catppuccin/assets/111578326/52f979a8-0620-4725-81a8-dc8d538c869d)

**Comment by sgoudham** [comment]
Closing as further discussion should take place in #2567.

We had some discussion in the discord around some of the worries/concerns noted by @unseen-ninja above. I'm personally happy to review this port as we decided to remove the instructions for the automatic CSS injector and that the manual instructions aren't that outlandish compared to some of our other ports.

**Reply by lemon-sh** [reply]
looks like you pinged the wrong 🍋 ^^"

**Reply by sgoudham** [reply]
Oh my god, so sorry!

**Reply by lemon-sh** [reply]
it's fine, don't worry about it
